### ISSUE TITLE
Dark mode with cupertino widget

### ISSUE THREAD

**Issue by SimonHausdorf** [jamesblasco/modal_bottom_sheet#138]
We are using `#000000` background in our dark mode theme. This leads to the following result

![image](https://user-images.githubusercontent.com/1446685/107136708-516b3e80-6938-11eb-96a4-9f79a6fe9051.png)

However, I was expecting something like this (the background color in iOS without bottom sheet open is black too, not gray):

![IMG_1043](https://user-images.githubusercontent.com/1446685/107136753-91cabc80-6938-11eb-988c-64d5e184e5d1.jpg)

I tried controlling the background with `backgroundColor` and `barrierColor` but this did not lead to the result expected.

**Comment by jamesblasco** [comment]
Working on it. Right now the only solution would be to change the background color of the previous route before the one is being pushed

**Comment by bierbaumtim** [comment]
I was playing around a bit with this issue and I found a solution that works for me. 
So if you want you can try it with my fork: https://github.com/bierbaumtim/modal_bottom_sheet

But as I said it works for me but there could be some issues and customizations aren't possible.

**Comment by jamesblasco** [comment]
Yes indeed @bierbaumtim has been doing amazing research on the topic. I am not completely familiar with BlendMode but I think using lighten and darken would be a better choice. 

As I can see in the native modal, it lightens up all the previous route while scrolling. You can see that with a backround completely dark.

Also for saturation, not all the page becomes grey, only primaryColor buttons, probably because they are disabled during an animation. We could implement something for this, but I don't think it would be useful if it is not integrated into the cupertino library itseld

**Comment by SimonHausdorf** [comment]
@bierbaumtim are you still interested in working on this? Your repository is a little bit behind. Would be great to see a PR from you to fix this issue.

**Comment by bierbaumtim** [comment]
@SimonHausdorf I created a new PR #231 which only contains the nessecary changes to create the discussed result.

**Comment by SimonHausdorf** [comment]
I tried the demo with your changes, settings the CupertinoNavigationBar to black. Theoretically it should turn into a dark gray correct? Similar to mail app in dark mode when composing a new email?


https://user-images.githubusercontent.com/1446685/160324925-00682160-c2c3-4b80-8ab4-ad33c98e8ece.mov

**Comment by bierbaumtim** [comment]
It only works if use CupertinoDynamicColor because it contains colors when CupertinoUserInterfaceLevel is elevated. With normal colors it would result in an undefined behaviour. 
Also the example does not support darkmode very well. I opened a second PR #232 which add darkmode support for the example.

**Comment by SimonHausdorf** [comment]
sorry to bother but I am still not able to get it work.

This is what I am trying to archive, black background / appBar and when bottom sheet opens it turns gray ish.

https://user-images.githubusercontent.com/1446685/161415355-2688b5df-5444-4068-8b3e-50649917b2d0.mov

**Comment by SimonHausdorf** [comment]
Ok, the magic was making use of `CupertinoTheme.of(context).scaffoldBackgroundColor,` I think this is the value your PR is changing?

Opening the bottom sheet looks good, but closing does not.

https://user-images.githubusercontent.com/1446685/161415895-d65c6be4-f2ef-4fb3-9358-b52ce23c5d4e.mov

**Comment by doug-orchard** [comment]
can confirm, this is still a problem
